test/test_bdecode.cpp

```diff
@@ -201,7 +201,8 @@ TORRENT_TEST(dict_null_key)
 	int ret = bdecode(b, b + sizeof(b)-1, e, ec);
 	TEST_EQUAL(ret, 0);
 	TEST_CHECK(e.dict_size() == 1);
-	bdecode_node d = e.dict_find(std::string("a\0b", 3));
+	std::string find_str("a\0b", 3);
+	bdecode_node d = e.dict_find(find_str);
```

Comment:
I think you can just skip the ``std::string`` entirely now. ``string_view`` should have a constructor from a string literal